pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -236,9 +251,23 @@ public String segmentCommit(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
     SegmentCompletionProtocol.Response response = segmentCompletionManager.segmentCommitStart(requestParams);
     if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
       // Get the segment and put it in the right place.
-      boolean success = uploadSegment(multiPart, instanceId, segmentName, false) != null;
-
-      response = segmentCompletionManager.segmentCommitEnd(requestParams, success, false);
+      URI segmentURI = uploadSegment(multiPart, instanceId, segmentName, false);
+      boolean success =  (segmentURI != null);
+      SegmentMetadataImpl segmentMetadata;
+      try {
+        segmentMetadata = extractMetadataFromLocalSegmentFile(segmentName, segmentURI.getPath());
```

Comment:
We should attempt to get metadata from the multi-part in this method as well. That way, we can add the server code to send multi-part form with metadata and everything will work. If metadata is not present in multi-part then we need to extract from local.
You can add this as a separate PR before moving on to other parts, or add it now, your choice.

Reply:
Actually, disregard this comment. If the segment is there, we don't need to push metadata, so you are good here.